tests/robustness/client/client.go

```diff
@@ -371,3 +372,73 @@ func toWatchEvent(event clientv3.Event) (watch model.WatchEvent) {
 	}
 	return watch
 }
+
+type ClientSet struct {
+	mux        sync.Mutex
+	idProvider identity.Provider
+	baseTime   time.Time
+
+	closed  bool
+	clients []*RecordingClient
+	reports []report.ClientReport
+}
```

Comment:
```suggestion
type ClientSet struct {
	idProvider identity.Provider
	baseTime   time.Time

	mux        sync.Mutex
	closed  bool
	clients []*RecordingClient
	reports []report.ClientReport
}
```

It good to have mutex over the variables it's meant to protect. We don't need to protect idProvider or baseTime as they are never changed.